core/src/commonMain/kotlin/com/revenuecat/purchases/kmp/PurchasesConfiguration.kt

```diff
@@ -32,6 +32,33 @@ public class PurchasesConfiguration private constructor(
                 "verificationMode=$verificationMode" +
                 ")"
 
+    internal fun storeKitVersionToUse(): StoreKitVersion {
+        var storeKitVersionToUse = this.storeKitVersion
+
+        if (this.purchasesAreCompletedBy is PurchasesAreCompletedBy.MyApp) {
+            storeKitVersionToUse = this.purchasesAreCompletedBy.storeKitVersion
+
+            if (this.storeKitVersion != StoreKitVersion.DEFAULT &&
+                storeKitVersionToUse != this.storeKitVersion) {
+                println(
+                    "Warning: The storeKitVersion in purchasesAreCompletedBy does not match the " +
+                            "provided storeKitVersion parameter. We will use the value found in " +
+                            "purchasesAreCompletedBy."
+                )
+            }
```

Comment:
> Question for my understanding: why do we have the separate storeKitVersion config option? Why not exclusively use the version set by PurchasesAreCompletedBy?

This is a _great_ question! 😄 The reason has to do with some implementation details with the StoreKit and PurchasesAreCompletedBy changes in the 5.X iOS SDK.

An optional `storeKit: StoreKitVersion` parameter was introduced to allow developers to specify which StoreKit version the SDK uses to process purchases. It's an optional parameter that defaults to StoreKitVersion.DEFAULT (which points to SK2) because ideally developers don't need to think about which SK version they're using (that's why they're using RC after all 😄).

On the other hand, we had to write a completely separate implementation of `PurchasesAreCompletedByMyApp` for SK2. When the SDK is initialized with `PurchasesAreCompletedBy==MyApp`, we need to know which `PurchasesAreCompletedByMyApp` implementation to wire up - the SK1 or SK2 implementation, so that's why providing a `StoreKitVersion` is mandatory when using `PurchasesAreCompletedByMyApp`. We're okay asking for a StoreKit version here because any developer who is already processing their own purchases should know the difference between SK1 and SK2, and which one they're using.

The reason that we can't exclusively use the version set by `PurchasesAreCompletedBy` is that we wanted people to be able to specify the StoreKit version without having to think about what `PurchasesAreCompletedBy` was. In retrospect, we might could have simplified this by adding `storeKitVersion` to `PurchasesAreCompletedBy.RevenueCat` and forgoing the `storeKitVersion` parameter entirely, but this is the API we settled on for this release 😅

Reply:
I've moved this from using `println` to `LogHandler` in https://github.com/RevenueCat/purchases-kmp/pull/139/commits/85fc14d4fd5eebd11a44289000dd1eaa5497c738. I used `[Purchases]` as the tag, do you know if there's a more appropriate value to use instead?

Reply:
Ah thanks for explaining! What do you think about doing this 👇  _only_ for purchases-kmp?
>we might could have simplified this by adding `storeKitVersion` to `PurchasesAreCompletedBy.RevenueCat` and forgoing the `storeKitVersion` parameter entirely

At first my opinion was that it's a no-go for consistency, but then I thought that possibly very few people will be using multiple RevenueCat SDKs, and comparing them. So curious what you think?

Reply:
Regarding the tag for logging: it's a convention to use the class name as a tag. So in this case that would be `PurchasesConfiguration`. **However**, I see that [in purchases-android we use `[Purchases]`](https://github.com/RevenueCat/purchases-android/blob/main/purchases/src/main/kotlin/com/revenuecat/purchases/common/logUtils.kt#L48), so I'd say let's use that here too. 😄

Reply:
I think my personal preference would be to keep it as is for the sake of API consistency across the hybrids. It's true that developers shouldn't be using multiple SDKs, but I'm afraid that allowing the APIs to drift apart could cause future development across the different hybrids to be more difficult.